### PR TITLE
Min js

### PR DESCRIPTION
**Pull request by tjikkun**
This is an idea to minify js files on build, to allow for unminified source js files in the repo, while keeping the benefits of minified js in the binary.
Please let me know if generally speaking you would like such a thing.

### CODE REVIEW

File: pdns/dnsdistdist/html/index.html [hunk 1]
```diff
@@ -1,50 +1,47 @@
...
+    top: 0;
+    bottom: 0;
+    width: 40px;
+    }
+
+    table.border, table.border>tbody>tr>td {
+    border: 1px solid black;
+    border-collapse: collapse;
+    }
+  </style>
```
Comment (cmouse) [inline pdns/dnsdistdist/html/index.html]: shouldn't this be in external css file instead?

**Comment by ahupowerdns** [comment]
THANK YOU! I like it when issues find our way as pull requests. Very good. A few questions, this pull request makes some CSS/HTML changes, was that intentional? Are they all clear improvements? Secondly, we have a new dependency now on 'uglify', is this is a commonly available tool?

File: pdns/dnsdistdist/html/index.html (hunk 1 above)
Comment (tjikkun) [inline pdns/dnsdistdist/html/index.html]: Why? I would argue to put everything in the html file, it gives no benefits to load seperate files
Anyway in this commit I just moved them from body to head

**Comment by tjikkun** [comment]
Well the html file had some validation errors that are fixed by the first commit, I guess that should be a seperate pull request. No real changes there, just bugfixes. (closing tag for head for example)
With regards to uglify dep, I am not really sure. If you decide something like this is good it would be easy to make it a configure option.

**Comment by ahupowerdns** [comment]
Ok - I don't want to impose an 'uglify' dependency on everyone building dnsdist. The "best" way to do this would be to ship pre-minified files in the tarball and have a Makefile rule that can minimize the unminified files for you if you change them. Compare 'ragel' in main PowerDNS.  This means that the minification should not happen in the incbin script. Thoughts?

**Comment by zeha** [comment]
For Debian purposes, having a new bundle.js is a nuisance (can't use it at all but have to rebuild at build time). Shipping the "official" pre-minified JS files plus their respective sources would be a lot easier.

**Comment by tjikkun** [comment]
@zeha Just for my information: Why is that different? How do you verify the prebuilt js?

**Comment by zeha** [comment]
By hashing the minified and source files. I think lintian has that (partially) automated. Even if there's no automation, I can't manually verify what's part of some bundle-min.js.

**Comment by tjikkun** [comment]
So Debian has a list of minified js files and their hashes?

**Comment by zeha** [comment]
Probably hidden somewhere in the lintian source.

**Comment by tjikkun** [comment]
@zeha I am still trying to understand how (manually) verifying by comparing hashes to a known source is easier then just rebuilding the source.

In my opinion it would be a good idea to also minify local.js, but you cannot verify that apart from rebuilding.

What do you propose as a better solution?

**Comment by zeha** [comment]
If there all source files shipped, I don't /really/ care. It'd let me avoid the uglifyjs build-dependency tho.

**Comment by Habbie** [comment]
This PR seems stuck. What can I do to move it along?

**Comment by tjikkun** [comment]
A time-machine or cloning device would be nice. I just need to find some time but I have been busy and not feeling too well in between. On a more serious note what is needed for Fedora is the ability to rebuild everything (so that includes js) and it would seem that from a Debian perspective that is not really needed.

As for bundling I would lean towards it, but not very strongly.

For the source javascripts it would be nice to know what versions are used.

I expect to have some time to work on this this Wednesday. Sorry it is taking so long.

**Comment by tjikkun** [comment]
New try:
 Add src JS files

The reason for this is to make downstreams like Fedora happy. This makes
checking those files for changes against upstream for example easier.

In the progress I updated the js-libraries so I know which versions are used
for both src and minified version. This is noted in the file src_js/SOURCES

Also I deleted jsrender.js and purl.js because the were unused.

Finally I added a Makefile option to build the minified js files yourself if
you want to do so.

**Comment by Habbie** [comment]
@zeha if you say yes I will merge this. I am not happy with it but I want this over with for now :)

**Pull request merged**